src/main/kotlin/lotto/application/LottoService.kt

```diff
@@ -0,0 +1,31 @@
+package lotto.application
+
+import lotto.domain.Lotto
+import lotto.domain.LottoNumberGenerator
+import lotto.domain.Rank
+
+class LottoService(
+    val lottos: List<Lotto>,
+) {
+    fun matchWinningNumber(winningNumbers: List<Int>): Map<Rank, Int> {
+        val matchResult = mutableMapOf<Rank, Int>()
+        lottos
+            .map { it.matchWinningNumber(winningNumbers) }
+            .map { Rank.valueOf(it) }
+            .forEach { matchResult[it] = matchResult.getOrDefault(it, 0) + INCREASE }
+        return matchResult.toMap()
```

Comment:
이 부분은 Kotlin 확장함수인 groupingBy와 eachcount의 조합으로 아래처럼 쓸 수도 있습니다 🙂
```suggestion
        return lottos
            .map { it.matchWinningNumber(winningNumbers) }
            .map { Rank.valueOf(it) }
            .groupingBy { it }
            .eachCount()
```

Reply:
groupingBy와 eachCount는 처음 접해보는데 신기하네요ㅋㅋㅋ
한번 알아봐야겠어요.
코드가 훨씬 더 깔끔해졌네요 !! 👍👍👍😄